src/org/zaproxy/zap/extension/ascanrulesAlpha/SQLInjectionMsSQL.java

```diff
@@ -0,0 +1,319 @@
+	 */
+	private boolean debugEnabled = log.isDebugEnabled();
+
+	@Override
+	public int getId() {
+		return 40027;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString("ascanalpha.sqlinjection.mssql.name");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return dependency;
+	}
+
+	@Override
+	public boolean targets(TechSet technologies) {
+		return technologies.includes(Tech.MsSQL);
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString("ascanalpha.sqlinjection.mssql.desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString("ascanalpha.sqlinjection.mssql.soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString("ascanalpha.sqlinjection.mssql.refs");
+	}
+
+	@Override
+	public void init() {
+		if ( this.debugEnabled ) log.debug("Initialising");
+
+		//set up what we are allowed to do, depending on the attack strength that was set.
+		if ( this.getAttackStrength() == AttackStrength.LOW ) {
+			doTimeBased=true; doTimeMaxRequests=3;
+		} else if ( this.getAttackStrength() == AttackStrength.MEDIUM) {
+			doTimeBased=true; doTimeMaxRequests=6;
+		} else if ( this.getAttackStrength() == AttackStrength.HIGH) {
+			doTimeBased=true; doTimeMaxRequests=12;
+		} else if ( this.getAttackStrength() == AttackStrength.INSANE) {
+			doTimeBased=true; doTimeMaxRequests=100;
+		}
+
+		// Read the sleep value from the configs
+		try {
+			this.sleep = this.getConfig().getInt("rules.common.sleep", 5);
+		} catch (ConversionException e) {
+			log.debug("Invalid value for 'rules.common.sleep': " + this.getConfig().getString("rules.common.sleep"));
+		}
+		if ( this.debugEnabled ) {
+			log.debug("Sleep set to " + sleep + " seconds");
+		}
+	}
+
+	/**
+	 * scans for SQL Injection vulnerabilities, using MsSQL specific syntax.  If it doesn't use specifically MsSQL syntax, it does not belong in here, but in TestSQLInjection
+	 */
+	@Override
+	public void scan(HttpMessage originalMessage, String paramName, String paramValue) {
+
+		try {
+			//Timing Baseline check: we need to get the time that it took the original query, to know if the time based check is working correctly..
+			HttpMessage msgTimeBaseline = getNewMsg();
+			long originalTimeStarted = System.currentTimeMillis();
+			try {
+				sendAndReceive(msgTimeBaseline, false); //do not follow redirects
+				}
+			catch (java.net.SocketTimeoutException e) {
+				//to be expected occasionally, if the base query was one that contains some parameters exploiting time based SQL injection?
+				if ( this.debugEnabled ) log.debug("The Base Time Check timed out on ["+msgTimeBaseline.getRequestHeader().getMethod()+"] URL ["+msgTimeBaseline.getRequestHeader().getURI().getURI()+"]");
+			}
+			long originalTimeUsed = System.currentTimeMillis() - originalTimeStarted;
```

Comment:
The messages have the elapsed time, which is calculated closer to the wire (e.g. it's not affected by other code that processes the message prior being sent, like scripts, authentication and other custom listeners...).